src/shared/yagna/adapters/market-api-adapter.ts

```diff
@@ -23,7 +42,7 @@ export class MarketApiAdapter implements MarketApi {
   async publishDemandSpecification(demand: DemandSpecification): Promise<Demand> {
     const idOrError = await this.yagnaApi.market.subscribeDemand(this.buildDemandRequestBody(demand.prototype));
 
-    if (typeof idOrError !== "string") {
+    if (idOrError && typeof idOrError !== "string") {
```

Comment:
I wasn't able to reproduce the issue where this would happen. I must've added it due to a failing test. Once I removed this change everything still works including the unit tests so we're good here.